app/component/itinerary/Legs.js

```diff
@@ -287,7 +287,7 @@ export default class Legs extends React.Component {
         <WalkLeg
           index={numberOfLegs}
           leg={compressedLegs[numberOfLegs - 1]}
-          previousLeg={compressedLegs[numberOfLegs - 2]}
+          previousLeg={compressedLegs[numberOfLegs - 1]}
```

Comment:
`compressedLegs[numberOfLegs - 1]` is repeated four times - it could be refactored into an appropriately named const